tests/test_landing_pad_tracking.py

```diff
@@ -0,0 +1,146 @@
+        obj2 = object_in_world.ObjectInWorld.create(-1,-1,0)[1]
+        assert not landing_pad_tracking.LandingPadTracking._LandingPadTracking__is_similar(obj1, obj2, DISTANCE_SQUARED_THRESHOLD)
+        obj1 = object_in_world.ObjectInWorld.create(0,0,0)[1]
+        obj2 = object_in_world.ObjectInWorld.create(1,1,0)[1]
+        assert not landing_pad_tracking.LandingPadTracking._LandingPadTracking__is_similar(obj1, obj2, DISTANCE_SQUARED_THRESHOLD)
+        obj1 = object_in_world.ObjectInWorld.create(0,0,0)[1]
+        obj2 = object_in_world.ObjectInWorld.create(0.5,0.5,0)[1]
+        assert landing_pad_tracking.LandingPadTracking._LandingPadTracking__is_similar(obj1, obj2, DISTANCE_SQUARED_THRESHOLD)
+        obj1 = object_in_world.ObjectInWorld.create(0,0,0)[1]
+        obj2 = object_in_world.ObjectInWorld.create(-0.5,-0.5,0)[1]
+        assert landing_pad_tracking.LandingPadTracking._LandingPadTracking__is_similar(obj1, obj2, DISTANCE_SQUARED_THRESHOLD)
+        obj1 = object_in_world.ObjectInWorld.create(0,0,0)[1]
+        obj2 = object_in_world.ObjectInWorld.create(2,2,0)[1]
+        assert not landing_pad_tracking.LandingPadTracking._LandingPadTracking__is_similar(obj1, obj2, DISTANCE_SQUARED_THRESHOLD)
+        obj1 = object_in_world.ObjectInWorld.create(0,0,0)[1]
+        obj2 = object_in_world.ObjectInWorld.create(-2,-2,0)[1]
+        assert not landing_pad_tracking.LandingPadTracking._LandingPadTracking__is_similar(obj1, obj2, DISTANCE_SQUARED_THRESHOLD)
+
+    def test_run_with_empty_detections_list(self, tracker: landing_pad_tracking.LandingPadTracking):
+        """
+        Test run method with empty detections list
+        """
+        success_flag, result = tracker.run([])
+        assert not success_flag
+        assert result is None
+
+    def test_run_single_input(self, tracker: landing_pad_tracking.LandingPadTracking, detections1: "list[object_in_world.ObjectInWorld]"):
+        """
+        Test run with only 1 input
+        """
+        success_flag, result = tracker.run(detections1)
+        assert success_flag
+        assert result == detections1[2]
+        assert tracker._LandingPadTracking__unconfirmed_positives == [detections1[2], detections1[1], detections1[4], detections1[0], detections1[3]]
+
+    def test_run_single_input_similar_detections(self, tracker: landing_pad_tracking.LandingPadTracking, detections1: "list[object_in_world.ObjectInWorld]"):
+        """
+        Test run with only 1 input where 2 landing pads are similar
+        """        
+        detections1[1] = object_in_world.ObjectInWorld.create(0.5, 0.5, 4)[1]
+        success_flag, result = tracker.run(detections1)
+        print(detections1[2].spherical_variance, result.spherical_variance)
+        assert success_flag
+        assert result == detections1[2]
+        assert tracker._LandingPadTracking__unconfirmed_positives == [detections1[2], detections1[1], detections1[4], detections1[3]]
```

Comment:
Place right side values into variables. Format of test should be:
* Setup: Sets up class/function to be tested and expected values.
* Run: Run the method/function.
* Test: Assertions.